api/src/main/java/org/apache/iceberg/types/Type.java

```diff
@@ -45,7 +46,8 @@ enum TypeID {
     DECIMAL(BigDecimal.class),
     STRUCT(StructLike.class),
     LIST(List.class),
-    MAP(Map.class);
+    MAP(Map.class),
+    VARIANT(VariantLike.class);
```

Comment:
This also needs to update the compatibility map in `Schema`: https://github.com/apache/iceberg/blob/main/api/src/main/java/org/apache/iceberg/Schema.java#L57-L58

Reply:
This also needs to update `Transforms` to not allow binding to a `VariantType` and should also update the type-specific constructors or factory methods to reject `VariantType`.

Reply:
`PartitionSpec` and `SortOrder` builders should reject `VariantType` too, right?